q2_aldex2/_visualizer.py

```diff
@@ -0,0 +1,134 @@
+        # close current plot, start one for MA
+        plt.close()
+
+        # base effect plot to build on
+        plt.scatter(x="rab.all", y="diff.btw", data=table, color="grey", s = 6)
+
+        # colour for significant points
+        plt.scatter(x="rab.all", y="diff.btw", data=table[called], color="red", s = 6)
+
+        # change titles and labels
+        plt.suptitle('MA plot')
+        plt.xlabel('Relative Abundance')
+        plt.ylabel('Difference')
+
+        img_ma = os.path.join(output_dir, "ma_plot.png")
+        plt.savefig(img_ma)
+
+        ########################################################################
+        # close current plot, start one for volcano
+        plt.close()
+
+        # base effect plot to build on
+        plt.scatter(x="diff.btw", y="we.eBH", data=table, color="grey", s = 6)
+
+        # colour for significant points
+        plt.scatter(x="diff.btw", y="we.eBH", data=table[called], color="red", s = 6)
+
+        # change titles and labels
+        plt.suptitle('Volcano plot')
+        plt.xlabel('Difference')
+        plt.ylabel('Q score')
+
+        img_volcano = os.path.join(output_dir, "volcano_plot.png")
+
+        # change p values to log scale
+        plt.yscale('log')
+        # get min and max to plot for p values
+        minimum = table["we.eBH"].min()
+        maximum = table["we.eBH"].max()
+        plt.ylim([minimum, maximum])
+
+        # plot line where cutoff is located
+        plt.plot([table["diff.btw"].min(), table["diff.btw"].max()], [threshold,threshold], color = "grey", linestyle='dashed', linewidth=1)
+
+        plt.savefig(img_volcano)
+
+        ########################################################################
+        # close current plot, start one for effect vs q score
+        plt.close()
+
+        # base effect plot to build on
+        plt.scatter(x="effect", y="we.eBH", data=table, color="grey", s = 6)
+
+        # colour for significant points
+        plt.scatter(x="effect", y="we.eBH", data=table[called], color="red", s = 6)
+
+        # change titles and labels
+        plt.suptitle('Effect size vs q score')
+        plt.xlabel('Effect size')
+        plt.ylabel('Q score')
+
+        # change p values to log scale
+        plt.yscale('log')
+        # get min and max to plot for p values
+        minimum = table["we.eBH"].min()
+        maximum = table["we.eBH"].max()
+        plt.ylim([minimum, maximum])
+
+        # plot line where cutoff is located
+        plt.plot([table["effect"].min(), table["effect"].max()], [threshold,threshold], color = "grey", linestyle='dashed', linewidth=1)
+
+        img_effect = os.path.join(output_dir, "effect_q_plot.png")
+
+        plt.savefig(img_effect)
+
+        # get path of index file
+        index = os.path.join(TEMPLATES, 'index.html')
+
+        # plot_name displays whether effect or MA
+        q2templates.render(index, output_dir, context={'plot_name': type})
```

Comment:
this doesn't look right ... `type` is a reserved keyword.   Are you sure you didn't mean to put in `test`?

Reply:
yup, good catch! updated.